src/Protocol/ActivityPub/Receiver.php

```diff
@@ -1851,7 +1851,7 @@ private static function extractAlternateUrl(array $urls, string $id): string
 				continue;
 			}
 
-			if ($mediatype == 'text/html' && $href != $id) {
+			if ($mediatype == 'text/html' && ($href != $id || $alternateUrl == '')) {
```

Comment:
```suggestion
			if ($mediatype === 'text/html' && ($href !== $id || $alternateUrl === '')) {
```

That makes it typesafe during checks

Reply:
Changed